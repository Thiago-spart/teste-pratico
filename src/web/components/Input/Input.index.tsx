/* eslint-disable import/no-default-export */
/* eslint-disable @typescript-eslint/no-explicit-any */
import { forwardRef } from "react";

import type { CustomInputProps } from "./Input.types";

import * as S from "./Input.styled";

const Input = (
	{ name, label, error = null, ...rest }: CustomInputProps,
	ref: any,
) => {
	return (
		<S.Container error={Boolean(error)}>
			<S.CustomLabel htmlFor={name}>{label}</S.CustomLabel>
			<S.InputStyle name={name} id={name} ref={ref} {...rest} />
		</S.Container>
	);
};

export default forwardRef(Input);
