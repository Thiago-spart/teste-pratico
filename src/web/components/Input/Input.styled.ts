import styled from "styled-components";

import { fonts } from "styles/fonts";

interface ContainerProps {
	error: boolean;
}

export const Container = styled.div<ContainerProps>`
	display: flex;
	align-items: flex-start;
	justify-content: flex-start;
	gap: 1.6rem;
	width: 100%;
	flex-direction: column;
	/* color: ${({ error }) => error && "var(--danger)"}; */

	/* input {
		border-color: ${({ error }) => (error ? "var(--danger)" : "var(--dark)")};
	} */
`;

export const CustomLabel = styled.label`
	${fonts.text4}
	font-weight: 400;
`;

export const InputStyle = styled.input`
	color: var(--text-color);
	border: 1px solid var(--input-border-color);
	border-radius: 0.8rem;
	padding: 2.4rem;
	width: 100%;
	transition: 0.2s;

	&::-webkit-outer-spin-button,
	&::-webkit-inner-spin-button {
		-webkit-appearance: none;
		margin: 0;
	}

	&[type="number"] {
		-moz-appearance: textfield;
	}

	&:focus-visible {
		outline: none;
		box-shadow: rgba(0, 0, 0, 0.45) 0px 25px 20px -20px;
	}
`;
