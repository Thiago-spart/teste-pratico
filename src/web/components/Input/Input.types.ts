import type { InputHTMLAttributes } from "react";
import type { FieldError } from "react-hook-form";

export interface CustomInputProps
	extends InputHTMLAttributes<HTMLInputElement> {
	name: string;
	label?: string;
	error?: FieldError | null;
}
