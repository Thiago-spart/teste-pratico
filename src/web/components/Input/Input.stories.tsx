/* eslint-disable import/no-default-export */
/* eslint-disable import/exports-last */
import type { ComponentStory, Meta } from "@storybook/react";

import Input from "./Input.index";

const meta: Meta<typeof Input> = {
	title: "Componentes/Input",
	component: Input,
};

const Template: ComponentStory<typeof Input> = props => <Input {...props} />;

export const Default = Template.bind({});
Default.args = {
	placeholder: "Text Input",
	label: "Text Input",
};

export default meta;
