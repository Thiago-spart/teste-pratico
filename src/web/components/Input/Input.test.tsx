import { composeStories } from "@storybook/testing-react";
import { render, screen } from "@testing-library/react";

import userEvent from "@testing-library/user-event";

import * as stories from "./Input.stories";

const { Default } = composeStories(stories);

describe("Button component", () => {
	it("should match snapshot", () => {
		const { container } = render(<Default />);

		expect(container).toMatchSnapshot();
	});

	it("should be filled up", () => {
		render(<Default />);

		const inputElement = screen.getByPlaceholderText("Text Input");

		userEvent.type(inputElement, "London");

		expect(screen.findByText("London")).toBeTruthy();
	});
});
