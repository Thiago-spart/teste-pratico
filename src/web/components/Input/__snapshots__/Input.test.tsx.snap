// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match snapshot 1`] = `
<div>
  <div
    class="sc-bczRLJ gVYvVB"
  >
    <label
      class="sc-gsnTZi gTeWuC"
    >
      Text Input
    </label>
    <input
      class="sc-dkzDqf XEbHD"
      placeholder="Text Input"
    />
  </div>
</div>
`;
